bin/civi-test-run

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+set -e
+
+function show_help() {
+  echo "Execute the CiviCRM test suites"
+  echo "usage: $0 <build-name> <civi-ver> <junit-output-dir> <test-type>"
+}
+
+if [ -z $1 ]; then
+  echo "No Build Name passed in"
+  echo
+  show_help
+  exit 2
+fi
+
+if [ -z $2 ]; then
+  echo "No CiviCRM Version passed in"
+  echo
+  show_help
+  exit 2
+fi
+
+if [ -z $3 ]; then
+  echo "No JUNIT Dir passed in"
+  echo
+  show_help
+  exit 2
+fi
+
+if [ -z $4 ]; then
+  echo "No Test type passed in"
+  echo
+  show_help
+  exit 2
+fi
+
+## Determine the absolute path of the directory with the file
+## absdirname <file-path>
+function absdirname() {
+  pushd $(dirname $0) >> /dev/null
+    pwd
+  popd >> /dev/null
+}
+
+BINDIR=$(absdirname "$0")
+PRJDIR=$(dirname "$BINDIR")
+BLDNAME=$1
+CIVIVER=$2
+BLDDIR="$PRJDIR/build/$BLDNAME"
+JUNITDIR=$3
+TESTTYPE=$4
+EXITCODES=
+GUARD=
+
+#################################################
+## Helpers
+function fatal() {
+  echo "$@"
+  exit 1
+}
+
+## RUN PHP Unit tests
+function runPHPUnit() {
+  case "$TESTTYPE" in
+  phpunit-api*) TESTS="api_v3_AllTests" EXITCODE="$EXITCODES phpunit API" ;;
+  phpunit-civi*) TESTS="Civi\\AllTests" EXITCODE="$EXITCODES phpunit Civi" ;;
+  phpunit-crm*) TESTS="CRM_AllTests" EXITCODE="$EXITCODES phpunit CRM" ;;
+  *) "CRM_AllTests api_v3_AllTests Civi\\AllTests" EXITCODE="$EXITCODES phpunit" ;;
+  esac
+
+  if $GUARD phpunit-each --exclude-group ornery "$BLDDIR/sites/all/modules/civicrm" "$JUNITDIR" $TESTS ; then
+    EXITCODES=$EXITCODE
+  fi
+  phpunit-xml-cleanup "$JUNITDIR"/*.xml
+}
+
+## Run Karma tests if requested Javascript
+if [ $TESTTYPE = "karma" ] || [ $TESTTYPE = "all" ] ; then
+  $GUARD pushd "$BLDDIR/sites/all/modules/civicrm"
+    ## Work-around: ensure pradmin user has sa contact
+    drush -u pradmin cvapi contact.get id=user_contact_id
```

Comment:
@totten this account may not always be set, so not sure what to do here

Reply:
@seamuslee001 If we moved `civi-test-run` into a subcommand of `civibuild` (eg `civibuild test dmaster ...` -- stored in `src/command/test-run.sh`), then `$ADMIN_USER` would be available implicitly, but that might be a bit more work to make it a subcommand.

This might work as a drop-in change:

```
ADMIN_USER=$(cv ev 'echo $GLOBALS['_CV']['ADMIN_USER'];')
drush -u "$ADMIN_USER" cvapi contact.get id=user_contact_id
```

or maybe this:

```
cv ev 'CRM_Utils_System::synchronizeUsers();'
```